### PR TITLE
[FEATURE]Migrating all CD pipelines to Ninja build + fix cu112 CD pipeline

### PR DESCRIPTION
**Pull request by access2rohit**
## Description ##
Migrates all cd build to `ninja` from `make` 
fixes CUDA11.2 pipeline for `v1.x`.
updates docker test images to ubuntu18 and fixes depedencies. 
Fixes build issues with cmake 
provides correct path for some make builds
Makes CD generated binaries compliant with Apache

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

## Testing ##
CD pipeline is working: https://jenkins.mxnet-ci.amazon-ml.com/job/restricted-mxnet-cd/job/mxnet-cd-pipeline-1.x-rohit/

### CODE REVIEW

File: cmake/Modules/FindCUDNN.cmake [hunk 1]
```diff
@@ -23,7 +23,7 @@ find_path(CUDNN_INCLUDE cudnn.h
   PATHS ${CUDNN_ROOT} $ENV{CUDNN_ROOT}
   DOC "Path to cuDNN include directory." )
 
-find_library(CUDNN_LIBRARY NAMES libcudnn.so cudnn.lib # libcudnn_static.a
+find_library(CUDNN_LIBRARY NAMES libcudnn.so cudnn.lib cudnn
```
Comment (leezu) [inline cmake/Modules/FindCUDNN.cmake:26]: Can you explain why this change is needed?

Comment (access2rohit) [reply cmake/Modules/FindCUDNN.cmake:26]: Added it while testing. Forgot to remove it. Will do

**Comment by access2rohit** [comment]
@leezu : can you review ?

**Comment by access2rohit** [comment]
@leezu @josephevans @mseth10 @Zha0q1 @waytrue17 can you review ?

**Review by leezu — APPROVED** [review]
Thank you. I think the main difference to the master branch now is that master CD builds on CentOS 7 whereas 1.x uses Ubuntu 14.04

**Review by Zha0q1 — APPROVED** [review]
LGTM thank you!

**Comment by access2rohit** [comment]
Latest CD successful build: https://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/restricted-mxnet-cd%2Fmxnet-cd-release-job-1.x-rohit/detail/mxnet-cd-release-job-1.x-rohit/58/pipeline

**Comment by access2rohit** [comment]
Apparently http://yann.lecun.com/exdb/mnist/ is down so added backup URL to`data.sh` to stablize unix-gpu CI.

**Comment by access2rohit** [comment]
@mxnet-bot run ci [unix-gpu]

**Comment by access2rohit** [comment]
Will disable flaky tests #20011 in this PR too if the current run fails.

**Review by Zha0q1 — APPROVED** [review]
LGTM! Thanks!
